pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -49,19 +57,79 @@ type PackageControllerClient struct {
 	// activeBundleTimeout is the timeout to activate a bundle on installation.
 	activeBundleTimeout time.Duration
 	valuesFileWriter    filewriter.FileWriter
+	// skipWaitForPackageBundle indicates whether the installer should wait
+	// until a package bundle is activated.
+	//
+	// Skipping the wait is desirable for full cluster lifecycle use cases,
+	// where resource creation and error reporting are asynchronous in nature.
+	skipWaitForPackageBundle bool
 }
 
 type ChartInstaller interface {
 	InstallChart(ctx context.Context, chart, ociURI, version, kubeconfigFilePath, namespace, valueFilePath string, values []string) error
 }
 
+// ChartInstallationDeleter handles deleting chart installations.
+type ChartInstallationDeleter interface {
+	Delete(ctx context.Context, kubeconfigFilePath, installName, namespace string) error
+}
+
+// NewPackageControllerClientFullLifecycle creates a PackageControllerClient
+// for the Full Cluster Lifecycle controller.
+//
+// It differs because the CLI use case has far more information available at
+// instantiation, while the FCL use case has less information at
+// instantiation, and the rest when cluster creation is triggered.
+func NewPackageControllerClientFullLifecycle(logger logr.Logger, chartInstaller ChartInstaller, deleter ChartInstallationDeleter, kubectl KubectlRunner) *PackageControllerClient {
```

Comment:
This feels like it warrants its own file.

Reply:
Why?

I like having it close to the NewPackageControllerClient, as it makes it easier to compare the two functions and see how they're similar or different. Plus they're both creating the same object.